src/FormModel.php

```diff
@@ -50,6 +50,9 @@ public function isAttributeRequired(string $attribute): bool
             if ($validator instanceof Required) {
                 return true;
             }
+            if ($validator instanceof HtmlOptionsProvider && (bool)($validator->getHtmlOptions()['required'] ?? false)) {
```

Comment:
What's the purpose? Can we have required rule that's not an instance of `Required`?

Reply:
I want to try to decompose the `FormModel` and remove `is attribute required` method. Now it's heavy-coupled with `Required` validation rule.